src/constraints.jl

```diff
@@ -44,83 +44,81 @@ function addconstraints! end
 addconstraints!(model::ModelLike, funcs, sets) = addconstraint!.(model, funcs, sets)
 
 """
-## Modify Function
+    canset(model::ModelLike, ::ConstraintSet, ::Type{ConstraintIndex{F,S}})::Bool
 
-    canmodifyconstraint(model::ModelLike, c::ConstraintIndex{F,S}, ::Type{F})::Bool
-
-Return a `Bool` indicating whether the function in constraint `c` can be replaced by another function of the same type `F` as the original function.
-
-## Modify Set
-
-    canmodifyconstraint(model::ModelLike, c::ConstraintIndex{F,S}, ::Type{S})::Bool
-
-Return a `Bool` indicating whether the set in constraint `c` can be replaced by another set of the same type `S` as the original set.
-
-## Partial Modifications
-
-    canmodifyconstraint(model::ModelLike, c::ConstraintIndex, ::Type{M})::Bool where M<:AbstractFunctionModification
-
-Return a `Bool` indicating whether it is possible to apply a modification of type `M` to the function of constraint `c`.
-
-### Examples
-
-```julia
-canmodifyconstraint(model, c, ScalarConstantChange{Float64})
-```
+Return a `Bool` indicating whether the set in a constraint of type `F`-in-`S`
+can be replaced by another set of the same type `S` as the original set.
 """
-function canmodifyconstraint end
-canmodifyconstraint(model::ModelLike, c::ConstraintIndex, change) = false
+canset(model::ModelLike, ::ConstraintSet, type_of_constraint_set) = false
+# Note: the above method is deliberately under-typed to avoid method ambiguities
 
 """
-## Modify Function
+    set!(model::ModelLike, ::ConstraintSet, c::ConstraintIndex{F,S}, set::S)
 
-    modifyconstraint!(model::ModelLike, c::ConstraintIndex{F,S}, func::F)
-
-Replace the function in constraint `c` with `func`. `F` must match the original function type used to define the constraint.
-
-### Examples
-
-If `c` is a `ConstraintIndex{ScalarAffineFunction,S}` and `v1` and `v2` are `VariableIndex` objects,
-
-```julia
-modifyconstraint!(model, c, ScalarAffineFunction([v1,v2],[1.0,2.0],5.0))
-modifyconstraint!(model, c, SingleVariable(v1)) # Error
-```
-
-## Modify Set
-
-    modifyconstraint!(model::ModelLike, c::ConstraintIndex{F,S}, set::S)
-
-Change the set of constraint `c` to the new set `set` which should be of the same type as the original set.
+Change the set of constraint `c` to the new set `set` which should be of the
+same type as the original set.
 
 ### Examples
 
 If `c` is a `ConstraintIndex{F,Interval}`
 
 ```julia
-modifyconstraint!(model, c, Interval(0, 5))
-modifyconstraint!(model, c, NonPositives) # Error
+set!(model, ConstraintSet(), c, Interval(0, 5))
+set!(model, ConstraintSet(), c, NonPositives)    # Error
```

Comment:
`Nonpositives` is the type. Maybe use a scalar set instead so it's a bit more clear, e.g., `LessThan(3)`.